p2p/pkg/rpc/provider/service.go

```diff
@@ -45,6 +45,10 @@ type ProviderRegistryContract interface {
 	GetProviderStake(*bind.CallOpts, common.Address) (*big.Int, error)
 	MinStake(*bind.CallOpts) (*big.Int, error)
 	ParseProviderRegistered(types.Log) (*providerregistry.ProviderregistryProviderRegistered, error)
+	ParseUnstake(types.Log) (*providerregistry.ProviderregistryUnstake, error)
```

Comment:
Not sure why it is necessary to define Parse* methods in this interface?

Reply:
I need to parse the event from the receipt to make sure that the tx was executed successfully. If I don't declare that here, I cannot call the function